blocks/library/tweet/index.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * Internal dependencies
+ */
+import { registerBlock, query } from '../../api';
+import Sandbox from '../../../components/sandbox';
+import Button from '../../../components/button';
+import Placeholder from '../../../components/placeholder';
+
+const { prop } = query;
+
+registerBlock( 'core/tweet', {
+	title: wp.i18n.__( 'Tweet' ),
+	icon: 'twitter',
+
+	category: 'social',
+
+	attributes: {
+		url: prop( '*', 'innerHTML' ), // our html is just a div with the url in, for WP's oembed to process
+	},
+
+	edit: class extends wp.element.Component {
+		constructor() {
+			super( ...arguments );
+			this.fetchTweet = this.fetchTweet.bind( this );
+			this.state = {
+				url: this.props.attributes.url,
+				html: '',
+				error: false,
+				fetching: false,
+			};
+		}
+		doFetch( url, setAttributes, setState ) {
+			setState( { fetching: true, error: false } );
+			jQuery.ajax( {
+				type: 'GET',
+				dataType: 'jsonp',
+				data: {},
+				timeout: 5000,
```

Comment:
We need to impose a timeout, or the error handling simply does not work. It shouldn't be applied globally though ajaxSetup because this is specific to JSONP requests.

From http://api.jquery.com/jQuery.ajax/
> Note: This handler is not called for cross-domain script and cross-domain JSONP requests.

There doesn't seem to be a default timeout value.

The other way of dealing with errors for cross domain requests is a jQuery plugin, https://github.com/jaubourg/jquery-jsonp , but it doesn't seem to be maintained.